src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,30 +35,37 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${showedTodo?.id}`}
```

Comment:
If you don't have todo just don't render UI, as in your case you render it but with empty elements